src/routes/v2/address.ts

```diff
@@ -141,70 +136,75 @@ async function detailsBulk(
     }
 
     // Enforce no more than 20 addresses.
-    if (addresses.length > 20) {
-      res.json({
+    if (addresses.length > FREEMIUM_INPUT_SIZE) {
+      res.status(400)
+      return res.json({
         error: "Array too large. Max 20 addresses"
       })
     }
 
     logger.debug(`Executing address/details with these addresses: `, addresses)
 
-    // stub response object
-    let returnResponse: IResponse = {
-      status: 100,
-      json: {
-        error: ""
-      }
-    }
+    // Validate each element in the address array.
+    for(let i=0; i < addresses.length; i++) {
+      const thisAddress = addresses[i]
 
-    // Loop through each address and creates an array of insight requests to call in parallel
-    addresses = addresses.map(async (address: any, index: number) => {
       // Ensure the input is a valid BCH address.
       try {
-        BITBOX.Address.toLegacyAddress(address)
-      } catch (er) {
-        if (er.message.includes("Unsupported address format"))
-          returnResponse.status = 400
-        returnResponse.json = {
-          error: `Invalid BCH address. Double check your address is valid: ${address}`
+        BITBOX.Address.toLegacyAddress(thisAddress)
+      } catch (err) {
+        if (err.message.includes("Unsupported address format")) {
+          res.status(400)
+          return res.json({
+            error: `Invalid BCH address. Double check your address is valid: ${thisAddress}`
+          })
         }
-        return
       }
 
       // Prevent a common user error. Ensure they are using the correct network address.
-      const networkIsValid = routeUtils.validateNetwork(address)
+      const networkIsValid = routeUtils.validateNetwork(thisAddress)
       if (!networkIsValid) {
-        returnResponse.status = 400
-        returnResponse.json = {
-          error: `Invalid network. Trying to use a testnet address on mainnet, or vice versa.`
-        }
+        res.status(400)
+        return res.json({
+          error: `Invalid network for address ${thisAddress}. Trying to use a testnet address on mainnet, or vice versa.`
+        })
       }
+    }
 
-      return await detailsFromInsight(address, currentPage)
+    // Loops through each address and creates an array of Promises, querying
+    // Insight API in parallel.
+    addresses = addresses.map(async (address: any, index: number) => {
+      return detailsFromInsight(address, currentPage)
     })
 
-    // if any input errors return response
-    if (returnResponse.status !== 100) {
-      res.status(returnResponse.status)
-      return res.json(returnResponse.json)
-    }
+    // Wait for all parallel Insight requests to return.
+    let result: Array<any> = await axios.all(addresses)
 
-    const result: Array<any> = []
-    return axios.all(addresses).then(
-      axios.spread((...args) => {
-        args.forEach((arg: any) => {
-          arg.legacyAddress = BITBOX.Address.toLegacyAddress(arg.addrStr)
```

Comment:
Apparently no documentation on `axios.spread` exists, which is frustrating, but I'm familiar with the spread operator in `Array`. After doing some tests and analyzing the output data, I reached the conclusion that a spread operator isn't needed since an Array is being returned, and then a `for` loop is being used to iterate over the elements. 

I replaced this code with a `map` function that processes the metadata. There was redundant address processing both here and in `detailsFromInsight()`. I pulled that code out of `detailsFromInsight()` and added to this function.